snowflake/main.go

```diff
@@ -14,8 +14,19 @@
 
 package main
 
-import "antrea.io/theia/snowflake/cmd"
+import (
+	"embed"
+
+	"antrea.io/theia/snowflake/cmd"
+	"antrea.io/theia/snowflake/pkg/infra"
+)
+
+// Embed the udfs directory here because go:embed doesn't support embeding in subpackages
+
+//go:embed udf/*
+var udfFs embed.FS
 
 func main() {
+	infra.UdfFs = udfFs
```

Comment:
I'd still avoid doing something like this. It can be confusing to change the value of a global variable which is in a different package. Maybe check out what I did for database migrations as a reference: https://github.com/antrea-io/theia/tree/main/snowflake/database

Reply:
Thanks Antonin, I created udfs.go in a separate package like database migrations and also found some utils function like `WriteMigrationsToDisk` could be shared here, changed it to `WriteEmbedDirToDisk` and moved to `utils` package.